File: Array/2996. Smallest Missing Integer Greater Than Sequential Prefix Sum/missingInteger.java
class Solution {
    public int missingInteger(int[] nums) {
        int sum = nums[0];
        boolean prefix;
        for (int i = 1; i < nums.length; i++) {
            if (nums[i - 1] == nums[i] - 1) {
                sum += nums[i];
            } else {
                break;
            }
        }
        while (true) {
            prefix = true;
            for (int i = 0; i < nums.length; i++) {
                if (nums[i] == sum) {
                    sum++;
                    prefix = false;
                }
            }
            if (prefix == true) {
                break;
            }
        }
        return sum;
    }
}

File: Array/2996. Smallest Missing Integer Greater Than Sequential Prefix Sum/missingInteger.rs
pub struct Solution;

impl Solution {
    pub fn missing_integer(nums: Vec<i32>) -> i32 {
        if nums.is_empty() {
            return 1; // 如果数组为空，返回1
        }
        let mut sum = nums[0];
        let mut prefix;
        for i in 1..nums.len() {
            if nums[i - 1] == nums[i] - 1 {
                sum += nums[i];
            } else {
                break;
            }
        }
        loop {
            prefix = true;
            for &num in &nums {
                if num == sum {
                    sum += 1;
                    prefix = false;
                }
            }
            if prefix {
                break;
            }
        }
        sum
    }
}